### PR TITLE
Cura 7290 manual account sync

### PR DESCRIPTION
**Pull request by ninovanhooff**
Pending https://github.com/Ultimaker/Cura/pull/7675

### CODE REVIEW

File: cura/API/Account.py [hunk 1]
```diff
@@ -57,13 +74,54 @@ def __init__(self, application: "CuraApplication", parent = None) -> None:
...
+        :param service_name: A unique name for your service, such as `plugins` or `backups`
+        :param state: One of Account.SyncState
+        """
+
+        prev_state = self._sync_state
+
+        self._sync_services[service_name] = state
+
+        if any(val == self.SyncState.SYNCING for val in self._sync_services.values()):
+            self._sync_state = self.SyncState.SYNCING
```
[the remark this answers is not in the document]
Comment (konskarm) [reply cura/API/Account.py:107]: Whoops, mentioned the same thing twice.

File: plugins/Toolbox/src/CloudSync/CloudPackageChecker.py [hunk 2]
```diff
@@ -69,15 +84,22 @@ def _onUserPackagesRequestFinished(self, reply: "QNetworkReply", error: Optional
             if "errors" in json_data:
                 for error in json_data["errors"]:
                     Logger.log("e", "%s", error["title"])
+                    self._application.getCuraAPI().account.setSyncState(self.SYNC_SERVICE_NAME, "error")
                 return
             self._handleCompatibilityData(json_data["data"])
         except json.decoder.JSONDecodeError:
             Logger.log("w", "Received invalid JSON for user subscribed packages from the Web Marketplace")
 
+        self._application.getCuraAPI().account.setSyncState(self.SYNC_SERVICE_NAME, "success")
```
Comment (ChrisTerBeke) [inline plugins/Toolbox/src/CloudSync/CloudPackageChecker.py:93]: You probably wanna use Account.SyncState.SUCCESS here as well?

File: plugins/Toolbox/src/CloudSync/CloudPackageChecker.py [hunk 3]
```diff
@@ -69,15 +84,22 @@ def _onUserPackagesRequestFinished(self, reply: "QNetworkReply", error: Optional
             if "errors" in json_data:
                 for error in json_data["errors"]:
                     Logger.log("e", "%s", error["title"])
+                    self._application.getCuraAPI().account.setSyncState(self.SYNC_SERVICE_NAME, "error")
```
Comment (ChrisTerBeke) [inline plugins/Toolbox/src/CloudSync/CloudPackageChecker.py:87]: Account.SyncState.ERROR?

File: plugins/Toolbox/src/CloudSync/CloudPackageChecker.py [hunk 4]
```diff
@@ -61,6 +75,7 @@ def _onUserPackagesRequestFinished(self, reply: "QNetworkReply", error: Optional
             Logger.log("w",
                        "Requesting user packages failed, response code %s while trying to connect to %s",
                        reply.attribute(QNetworkRequest.HttpStatusCodeAttribute), reply.url())
+            self._application.getCuraAPI().account.setSyncState(self.SYNC_SERVICE_NAME, "error")
```
Comment (ChrisTerBeke) [inline plugins/Toolbox/src/CloudSync/CloudPackageChecker.py:78]: Account.SyncState.ERROR?

File: cura/API/Account.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ninovanhooff) [reply cura/API/Account.py:107]: Ah yes, this got lost when I refactored the timer. Thanks.

File: resources/qml/Account/SyncState.qml [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+                onEntered: accountSyncButton.font.underline = true
+                onExited: accountSyncButton.font.underline = false
+            }
+        }
+    }
+
+    signal syncStateChanged(string newState)
+
+    onSyncStateChanged: {
+        if(newState == "syncing"){
```
[the remark this answers is not in the document]
Comment (ninovanhooff) [reply resources/qml/Account/SyncState.qml:85]: Great, I assumed this was not possible. I'll have a go at it

**Pull request merged**